Mathlib/RingTheory/Localization/Module.lean

```diff
@@ -31,15 +33,21 @@ open nonZeroDivisors
 
 section Localization
 
-variable {R : Type*} (Rₛ : Type*) [CommRing R] [CommRing Rₛ] [Algebra R Rₛ]
+variable {R : Type*} [CommRing R] (Rₛ : Type*) [CommRing Rₛ] [Algebra R Rₛ]
 
 variable (S : Submonoid R) [hT : IsLocalization S Rₛ]
 
 -- include hT
 
+section IsLocalizedModule
+
 section AddCommMonoid
 
-variable {M : Type*} [AddCommMonoid M] [Module R M] [Module Rₛ M] [IsScalarTower R Rₛ M]
+open Submodule
+
+variable {M M' : Type*} [AddCommMonoid M] [Module R M] [Module Rₛ M] [IsScalarTower R Rₛ M]
+  [AddCommMonoid M'] [Module R M'] [Module Rₛ M'] [IsScalarTower R Rₛ M'] (f : M →ₗ[R] M')
+  [IsLocalizedModule S f]
 
 theorem LinearIndependent.localization {ι : Type*} {b : ι → M} (hli : LinearIndependent R b) :
```

Comment:
Can this be obtained using `isLocalizedModule_of_isScalarTower` below?

Reply:
I don't see how? That one is about localization of an algebra, and doesn't apply to general modules.

Reply:
I guess you mean `LinearIndependent.of_isLocalizedModule` instead? It's possible, by introducing the following missing lemma (can be inserted just after `IsLocalizedModule.eq_iff_exists` in the other file):
```
variable (M) in
lemma isLocalizedModule_id (R') [CommRing R'] [Algebra R R'] [IsLocalization S R'] [Module R' M]
    [IsScalarTower R R' M] : IsLocalizedModule S (.id : M →ₗ[R] M) where
  map_units s := (Module.End_isUnit_iff _).mpr <| by
    convert_to Function.Bijective (algebraMap R' (Module.End R' M) (algebraMap R R' s))
    · ext; simp
    exact (Module.End_isUnit_iff _).mp ((IsLocalization.map_units R' s).map _)
  surj' m := ⟨(m, 1), one_smul _ _⟩
  exists_of_eq h := ⟨1, congr_arg _ h⟩
```
then this becomes
```
theorem LinearIndependent.localization {ι : Type*} {b : ι → M} (hli : LinearIndependent R b) :
    LinearIndependent Rₛ b :=
  have := isLocalizedModule_id S M Rₛ
  hli.of_isLocalizedModule Rₛ S .id
```